src/scene/procedural.js

```diff
@@ -35,6 +35,9 @@ import { Mesh } from './mesh.js';
 var primitiveUv1Padding = 4.0 / 64;
 var primitiveUv1PaddingScale = 1.0 - primitiveUv1Padding * 2;
 
+// cached mesh primitives
+var shapePrimitiveMap = new Map();
```

Comment:
What happens here if two instances of the engine running on the same page start creating primitives? Such as with [this app](http://playcanvas.github.io/#misc/multi-application.html).

Reply:
Oh actually, I see you pass in the graphics device so it's fine. Ignore. 😄